### PR TITLE
Rebuild

### PR DESCRIPTION
**Pull request by seanmcoleman**
## Remaining Scope
- [x] Configure logging (need to research PHP library best practices).
- [x] Default necessary config values to implicitly set environment variables.
- [x] Auto-apply config loaded from env variable.
- [x] Update license
- [x] Reconfigure entrypoint file from test.php
- [x] README
- [ ] Unit tests
- [ ] Test/adjust job function

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,31 +1,19 @@
...
     {
-      "name": "Nathaniel Schweinberg",
-      "email": "nathaniel@fightthecurrent.org"
+      "name": "Cronitor"
     }
   ],
   "require": {
-    "php": ">=5.4.0",
-    "anlutro/curl": "~1.4"
+    "php": ">=5.6.0",
```
Comment (seanmcoleman) [inline composer.json:16]: I intentionally kept the require PHP version low for compatibility.

File: lib/HttpClient.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    public function put($path, $body = array(), $params = array())
+    {
+        return $this->request($path, 'PUT', $params, $body);
+    }
+
+    private function request($path, $httpMethod, $params = array(), $body = null)
+    {
+        $options = array(
```
Comment (seanmcoleman) [inline lib/HttpClient.php:35]: I opted for the built-in cURL interface so we didn't need to require a dependency of another http library, and all the version maintenance that comes along with it.

File: lib/Monitor.php [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+            return false;
+        }
+
+        try {
+            $queryString = $this->buildPingQuery($params);
+            $client = $this->getPingClient($retryCount);
+            $response = $client->get("/?$queryString");
+            $responseCode = $response['code'];
+
+            if ($responseCode < 200 || $responseCode > 299) {
```
Comment (seanmcoleman) [inline lib/Monitor.php:90]: The Ruby library was checking a straight 200 response, but I see a 204 code. Regardless, checking against the full 200-range seems to be the best practice.

**Review by aflanagan — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -1,112 +1,192 @@
...
-You can use `Cronitor\Client` by itself:
+You can also use a YAML config file to manage all of your monitors (_see Create and Update Monitors section below_). The path to this file can be supplied using the enviroment variable `CRONITOR_CONFIG` or call `$cronitor.readConfig()`.
 
 ```php
-$monitorId = 'id of the monitor found on the cronitor dashboard';
-$client = new Cronitor\Client($monitorId);
-$client->run();
-// do some stuff...
-$client->complete();
+$cronitor.readConfig('./path/to/cronitor.yaml');
```
Comment (mrsimonbennett) [inline README.md:36]:
```php
$cronitor->
```

File: README.md [hunk 5]
```diff
@@ -1,112 +1,192 @@
...
-} catch (Exception $e){
-  // Do some other stuff, like logging perhaps?
-}
+require 'cronitor'
+
+$monitor = $cronitor->monitor('heartbeat-monitor');
+
+$monitor->pause(24) # pause alerting for 24 hours
+$monitor->unpause() # alias for .pause(0)
+$monitor->ok() # manually reset to a passing state alias for monitor.ping({state: ok})
```
Comment (mrsimonbennett) [inline README.md:78]:
```php
$cronitor->
```

File: lib/Cronitor.php [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+    public function validateConfig()
+    {
+        $this->applyConfig(true);
+    }
+
+    public function job($key, $callback)
+    {
+        $monitor = $this->monitor($key);
+        $series = microtime(true);
+        $monitor.ping(['state' => 'run', 'series' => $series]);
```
Comment (mrsimonbennett) [inline lib/Cronitor.php:127]:
```php
$cronitor->
```

File: lib/Cronitor.php [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+
+        try {
+            $callback();
+            $monitor.ping(['state' => 'complete', 'series' => $series]);
+        } catch (Exception $e) {
+            $message = $e->getMessage();
+            $truncatedMessage = $message.substr(abs(min(0, 1600 - strlen($message))));
+            $monitor.ping([
```
Comment (mrsimonbennett) [inline lib/Cronitor.php:135]:
```php
$cronitor->
```

File: lib/Cronitor.php [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+
+    public function job($key, $callback)
+    {
+        $monitor = $this->monitor($key);
+        $series = microtime(true);
+        $monitor.ping(['state' => 'run', 'series' => $series]);
+
+        try {
+            $callback();
+            $monitor.ping(['state' => 'complete', 'series' => $series]);
```
Comment (mrsimonbennett) [inline lib/Cronitor.php:131]:
```php
$cronitor->
```

File: lib/HttpClient.php (hunk 2 above)
Comment (mrsimonbennett) [reply lib/HttpClient.php:35]: Might be worth adding to required dev

```json
"ext-curl": "*",
```

File: lib/HttpClient.php [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    private function headers()
+    {
+        return array(
+            'Content-Type: application/json',
+            'Accept: application/json',
+            'User-Agent: cronitor-php',
+            "Cronitor-Version: $this->apiVersion",
+        );
```
Comment (mrsimonbennett) [inline lib/HttpClient.php:68]:
```suggestion
        return [
            'Content-Type: application/json',
            'Accept: application/json',
            'User-Agent: cronitor-php',
            "Cronitor-Version: $this->apiVersion",
        
       ];
```

**Review by mrsimonbennett — CHANGES_REQUESTED** [review]
Just some feedback :)

**Comment by seanmcoleman** [comment]
Thanks @mrsimonbennett! I'm closing this PR since it's orphaned/left-over, but have opened #19.

**Pull request closed without merging**